components/filter/filter-dimension-set-empty-state.js

```diff
@@ -0,0 +1,58 @@
+import { LitElement } from 'lit';
+
+/**
+ * A component to customize the empty state parameters for a particular filter-dimension-set.
+ * This component does not render anything, but instead gathers data needed for the d2l-filter.
+ */
+class FilterDimensionSetEmptyState extends LitElement {
+
+	static get properties() {
+		return {
+			/**
+			 * The href that will be used for the empty state action. When set with action-text, d2l-filter will render a link action.
+			 * @type {string}
+			 */
+			actionHref: { type: String, attribute: 'action-href' },
+			/**
+			 * The text that will be displayed in the empty state action. When set, d2l-filter renders a button action, or a link if action-href is also defined.
+			 * @type {string}
+			 */
+			actionText: { type: String, attribute: 'action-text' },
+			/**
+			 * REQUIRED: The text that is displayed in the empty state description
+			 * @type {string}
+			 */
+			description: { type: String }
+		};
+	}
+
+	constructor() {
+		super();
+		this.actionHref = '';
+		this.actionText = '';
+		this.description = '';
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		const changes = new Map();
+		changedProperties.forEach((oldValue, prop) => {
+			if (oldValue === undefined) return;
+
+			if (prop === 'actionHref' || prop === 'actionText' || prop === 'description') {
```

Comment:
So one thing you might have noticed about `d2l-filter-dimension-set` is that it doesn't list all properties here for simplicity.  We made the decision that, until we had a use-case for it, you don't get to switch your search type or whether the set is single-dimension-only after the fact. Do you know if the same would apply here?  Or does their use case call for different empty state text/action based on the search entered?

Reply:
Ah okay. I'll check, but as far as I know that is not a requirement and we could probably do it this way as well. In the case of the empty state though, these properties are just text values so it doesn't add much complexity to handle the change events so personally I see it as a nice-to-have.

But yeah, if we don't need that, then we could potentially remove the `d2l-filter-empty-state-change` events and handlers altogether.

Reply:
Yeah, up to you. Doesn't save a _whole_ lot, just trying to think of ways to keep filter simpler and the component wrapping less awkward.

Reply:
Although, yeah, if we could _also_ clean up the `slot` change events for it, and just say that you need to have it there from the get-go, I think that cleans up _a lot_.

Thinking out loud, since I'm still not all the way through the filter code - without the data events, would that make tying it back to the same `d2l-filter-diemnsion-set-data-change` event and `_handleDimensionDataChange` simpler?  Like when the slot changes for the search empty set, you'd:
```
this._dispatchEvent('d2l-filter-dimension-data-change', { dimensionKey: this.key, changes: new Map([['searchEmptyState', this.getSearchEmptyState()]]) });
```
Which would trigger `_handleDimensionDataChange` to set it, without any changes needed to that function because it doesn't need to cause a recount or a search or anything fancy?

Reply:
Yeah, I was doing something like this initially but I thought it was cleaner and easier to understand as a separate event and thought the data change handler was already doing a lot. I don't mind going back to that though if it's preferred and it would be a lot simpler without the empty state data changes.

I'll get back to you on whether those data change events are needed.

Reply:
> I don't mind going back to that though if it's preferred and it would be a lot simpler without the empty state data changes.

Yeah I'd be interested in seeing what it looks like, maybe as a separate PR/branch to compare easily

Reply:
So design doesn't have an explicit rule on search descriptions and we leave it up to the consumer on if they want to use something like `x returned no results` but there isn't any use case that requires it at the moment. It still could be nice to have for the future if it was needed at some point though. 

I'll create a separate branch and try your suggestion